_posts/2023-02-22-#25038.md

```diff
@@ -7,18 +7,234 @@ authors: [glozow]
+	fees required for Alice to replace B by:
+
+		(1) Attaching transaction(s) that descend from B and pay a feerate too low to fee-bump B through
+		   CPFP. For example, assuming the default descendant size limit is 101KvB and B is 1000vB paying a
+		feerate of 2sat/vB, adding a 100KvB, 2sat/vB child increases the cost to replace B by 200Ksat.
+
+		(2) Adding a high-fee descendant of B that also spends from a large, low-feerate mempool transaction,
+		   C. The child may pay a very large fee but not actually be fee-bumping B if its overall ancestor
+		feerate is still lower than B's individual feerate. For example, assuming the default ancestor size
+		limit is 101KvB, B is 1000vB paying 2sat/vB, and C is 99KvB paying 1sat/vB, adding a 1000vB child of
+		B and C increases the cost to replace B by 101Ksat.
+
+- Pinning through package limits: All transactions admitted to the mempool must stay within package limits; no ancestor
+  set or descendant set may exceed 25 transactions. Additionally, Rule 5 requires a replacement cause the eviction of
+more than 100 entries.
+
+	- When a candidate transaction would replace mempool entries, it may also decrease the descendant count of other
+	  mempool entries. Since ancestor/descendant limits are calculated prior to removing the would-be-replaced
+	transactions, they may be overestimated.
+	Also see [Single-Conflcit RBF Carve Out](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/mempool-limits.md#single-conflict-rbf-carve-out).
+
+	- If a transaction conflicts with multiple mempool entries, even if each entry is within the 25-descendant
+	  package limit, it is possible to reach the Rule 5 limit.
+
+- No increase in incentive compatibility required: Theoretically any replacement transaction should
+  confirm faster than the one it replaced; a rational user would not intentionally pay more fees to
+have their transaction confirm later. However, there is currently no rule to enforce this. As a
+result, all transactions signed with SIGHASH\_ANYONECANPAY are vulnerable to being replaced by a transaction
+that will confirm later than the original.
+
+	- [Rule 6](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/mempool-replacements.md#current-replace-by-fee-policy)
+	was originally enforcing this, as the "incentive compatibility score" was previously just the
+	feerate of the transaction. However, with the current ancestor set-based selection algorithm and
+	possible package topologies, this is difficult to precisely calculate short of building a block
+	template.
+
+	- We have discussed the difficulty of calculating "miner score" in a [previous review club meeting](/26152).
+	Also see discussion in [#23121](https://github.com/bitcoin/bitcoin/pull/23121) and
+	[#26451](https://github.com/bitcoin/bitcoin/pull/26451).
+
+- Descendants of BIP125 signaling transactions cannot be directly replaced: Any descendant of an
```

Comment:
> Descendants of BIP125 signaling transactions cannot be directly replaced:

I think you mean to say that they can't be directly replaced if they don't signal for RBF? Currently, it reads as if they can't be replaced at all.

Reply:
Blehhh, hard to word, got dizzy just trying to write it :joy: Hopefully clearer now